book/src/proposals/tower-lockout-slashing.md

```diff
@@ -0,0 +1,107 @@
+# Tower Lockout Slashing
+
+This design describes how slashing is implemented in the solana
+cluster.  Slashing is designed as a punishment for participants in
+the cluster that violate the rules of the protocol.  A proof of the
+violation is submitted to the cluster and if the proof is accepted,
+the stake that is delegated to the node that violated a rule is
+slashed by some percentage.
+
+## Votes
+
+Validators submit votes which contain a list of slots.  To be
```

Comment:
Could we expound on this a little bit? Specifically how the votes in replay are constructed. I think there may be an issue based on the current definition of vote slashing.

So to my understanding I think the proposed procedure for voting on a bank `B` was:

1) Add `B` to the current locktower state.
2) Find validator's vote state `v` in the bank state of the bank `B`.
3) Because we are voting on `B` we know that `B` must be descendant of all banks in the current locktower state `L`. Thus we know that all votes in `L` are ancestors of `B` and thus playable on `B`.
4) From 3) we then know it's safe to to take `L \ v` and replay it on `B`, submitting that resulting state as your vote for bank `B`.

Reply:
So point of clarification on step 4) illustrated by the below example:

A validator is alternating between two forks `A` and `B` as follows. 

The locktower state looks like:

1) `0, 1, 2, 3, ....17, 18, 19, 21` (Fork A)
2) `0, 1, 2, 3, ....17, 18, 19, 24` (Fork B)
3) `0, 1, 2, 3, ....17, 18, 19, 27` (Fork A)

Now on the third vote, would the validator would have to submit `0, 1, 2, 3, ....17, 18, 19, 21, 27`, even thought 27 has already popped off 21 in order to get the first 20 votes to match the bank state?

Reply:
@carllin it doesn't have to be exactly 20.  Just he lockout on the first vote should be > epoch

Reply:
@aeyakovenko, that seems to conflict with the statement in the document: `The first `N - new votes`, must match the VoteState after running the tower consensus algorithm with the new votes.` Could we clarify what votes needs to match / why it's important lockout on first vote > epoch?

Reply:
The timeout on slashing proofs is one epoch, so the proofs must cover that period.

Reply:
I think we need to first define in this document what it means for a slashing proof to be either "valid or timed out", so that there is intuition for this.

Reply:
check the latest.